### ISSUE TITLE
Default Language Context MAY -> SHOULD

### ISSUE THREAD

**Issue by r12a** [w3c/activitystreams#354]
This is a comment on https://github.com/w3c/activitystreams/pull/352 (raised here so that i18n folks are notified).

The section Default Language Context says:

> When using [[JSON-LD]] mechanisms to produce or consume Activity Streams 2.0 documents, the @language property MAY be used within a @context to identify the default language. This mechanism may not be understood by implementations that do not choose to process Activity Streams 2.0 documents using JSON-LD. 

We were previously explaining that associating language with strings is often necessary when it comes to presenting them to a user (eg. to prevent Chinese users being presented with Japanese font glyphs).  Could we either change MAY to SHOULD, or at least add a note to say that actually language information is quite important (since very few developers seem to recognise that)?

**Comment by akuckartz** [comment]
It is not necessary to provide that property in the _context_ to associate all strings with a language. But I agree that such a note might be helpfull.

**Comment by r12a** [comment]
Ah yes, i follow your reasoning. Perhaps a note then?

**Comment by evanp** [comment]
So, I think this is a valid Activity Streams document:

```json
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "@language": "fr",
  "type": "Note",
  "name": "Une note brève"
}
```

I vastly prefer this to the version with `@language` inside the `@context`, which seems kind of complex.

@jasnell If this is the case, maybe we can just recommend that documents SHOULD include the `@language` property and be done with it?

**Comment by jasnell** [comment]
While this is valid, having the `@language` here would have no effect when it comes to JSON-LD processing. The `@language` needs to either be in the `@context` or inside a "value object" like `"name": {'@language': 'en', '@value': 'Une note brève`}`. The latter is a form we don't want to promote and currently rule out in our syntax because of the additional parsing complexity is requires.

**Comment by evanp** [comment]
It's interesting how the JSON-LD playground handles it:

http://tinyurl.com/j3eww25

versus

http://tinyurl.com/jg53eyh

**Comment by evanp** [comment]
Accidentally closed

**Comment by jasnell** [comment]
Yeah, if you look at the normalized result for the [first](http://tinyurl.com/j3eww25), you'll see that the `@language` is lost entirely. That's because it is treated as a non-declared extension property that is simply ignored by the JSON-LD processing.

**Comment by akuckartz** [comment]
@lanthaler Do you have any suggestion or opinion regarding this issue ?

**Comment by sandhawke** [comment]
http://www.w3.org/2016/11/18-social-irc#T17-18-23

**Comment by sandhawke** [comment]
RESOLVED: Resolve issue #354 of Activity Streams with a SHOULD that points out the importance of explicitly marking up language of natural language properties if you know the language, and use such markup for some examples, and points out that it's not in every example because we want to avoid the copypaste EN everywhere thing
